website_sale_stock_provisioning_date/models/product_product.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2019 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    def _get_next_provisioning_date(self):
+        domain = [
+            ("company_id", "=", self.env.user.company_id.id),
```

Comment:
Is this the same way as Odoo uses in the stock info popup for sales?

Reply:
In the popup, the date that is shown is the expected date to deliver the products defined in the sale order.

![2020-09-22_11-17](https://user-images.githubusercontent.com/38267832/93902303-66b33980-fcc5-11ea-9b67-c950617d0258.png)

Reply:
Uhm, how about the expected one, how is computed?

Reply:
But the expected date is information related to the sales order and what we want is information related to a product. In this case, the nearest future date the product is expected to be received. In the same sale order. In the same sales order we could have several products with different expected reception dates.

Reply:
Yeah, but this computation is done taking the minimum expected date for each product. How that expected date is computed? We want the same algorithm (removing the customer lead time).

Reply:
That is in v13 and as I see in the code it is almos the same as the date of the sale order.
In v12 there is effective_date, but what it calculates is the minimum date of the done pickings related to the sale_order.
 I don't think those fields are useful for what we want.

Reply:
OK